server/js/src/main/scala/org/http4s/server/SecureSession.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.server
+
+import scodec.bits.ByteVector
+
+// TODO not sure about this representation
```

Comment:
If there's demand, and somebody knows how to parse that raw binary encoding into something intelligble. Unfortunately Node.js just gives us that.

Reply:
The representation is immutable, but not extensible.  As long as it's unlikely to be extended, what you've done is fine.  Otherwise, we want one of those "contraband" encodings, like the builders.

Reply:
Actually since I wrote this, I ended up implementing a similar wrapper in fs2.io. Maybe we can use that instead?
https://github.com/typelevel/fs2/pull/2510